dbms/src/Storages/Page/V3/Blob/BlobFile.cpp

```diff
@@ -41,7 +41,8 @@ BlobFile::BlobFile(
         getPath(),
         getEncryptionPath(),
         false,
-        /*create_new_encryption_info_*/ !file_in_disk.exists());
+        /*create_new_encryption_info_*/ !file_in_disk.exists(),
+        /*skip_encryption*/ file_provider->isKeyspaceEncryptionEnabled());
```

Comment:
```suggestion
        /*skip_encryption*/ file_provider->isKeyspaceEncryptionEnabled() // When keyspace encryption is enabled, we encrypt the page data instead of the whole BlobFile
    );
```